### DISCUSSION TITLE
Poetry version specification not respected

### DISCUSSION THREAD

**Discussion by bcapitanu-asapp** [renovatebot/renovate#32921] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

39.23.0

### Please tell us more about your question or problem

I have a Poetry dependency specified as:
```
httpx = "~0.27"
```
but renovate has created a PR to update it to `~0.28.0` which doesn't respect the constraint originally specified (`>=0.27.0,<0.28.0`)

![image](https://github.com/user-attachments/assets/1ea6d24a-b668-4531-8e4f-53302eab36d3)

Here's the renovate configuration I'm using:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>ASAPPinc/renovate:base"
  ],
  "packageRules": [
    {
      "groupName": "{{manager}} dependencies",
      "groupSlug": "{{manager}}",
      "separateMajorMinor": true,
      "matchPackageNames": [
        "*"
      ]
    }
  ],
  "enabledManagers": [
    "jsonnet-bundler",
    "poetry"
  ],
  "lockFileMaintenance": {
    "enabled": true
  },
  "schedule": "every weekday",
  "excludeCommitPaths": [
    "tanka/vendor/**"
  ]
}
```

**Comment by rarkins** [comment]
"Respect" does not mean "never update it". Renovate's job is to update dependencies. It's like complaining that a car wash cleaned your car.

If you mean "I want to stay on `~0.27` forever - or until I update it manually myself - then that's an entirely different and more reasonable question.

**Reply by bcapitanu-asapp** [reply]
Thanks for your answer --- I was imagining that Renovate understands the version constraints specified in the Poetry config file and only decides to offer an update if the update respects those constraints.  Maybe I misunderstood how it's supposed to work - apologies!

Is there a way I can configure Renovate to behave in that fashion?   What is your recommendation, given the desired behavior I'd like to see?  (and thank you again for taking the time to respond!)

**Reply by bcapitanu-asapp** [reply]
But yes, if the desired behavior I mention is not possible, then the fallback will be:
"I want to stay on ~0.27 forever - or until I update it manually myself"

**Reply by rarkins** [reply, marked as answer]
The downsides with this approach is that you'll be out of date and probably unaware, may not get vulnerability fixes, etc.

But you could try if `rangeStrategy=in-range-only` makes a positive difference to you

**Reply by bcapitanu-asapp** [reply]
Perfect!  Thank you so much for your help! ❤️
